source/System.cpp

```diff
@@ -240,7 +281,30 @@ void System::Load(const DataNode &node, Set<Planet> &planets)
 					}
 			}
 			else
-				fleets.emplace_back(fleet, child.Value(valueIndex + 1));
+			{
+				int period = 200;
+				int limit = LimitedEvents<Fleet>::NO_LIMIT;
+				int initialCount = 0;
+				string id;
+				bool ignoreEnemyStrength = false;
+
+				if(child.Size() > valueIndex + 1)
+					period = child.Value(valueIndex + 1);
+				if(child.HasChildren())
+					LoadFleet(child, period, limit, initialCount, id, ignoreEnemyStrength);
+
+				bool defaultFleetId = (limit >= 0 || initialCount > 0) && id.empty();
+				if(defaultFleetId)
+					id = value + "@" + name;
```

Comment:
in theory you could have two fleets with the exact same period, but they would be handled as the same fleet?

Reply:
That is intentional. If you don't give a fleet an id, then the limit is based on the fleet, not the individual fleet entry.

Also, multiple copies of the same fleet doesn't really work as (some?) people think in Endless Sky right now. If you do this:

```
    system whatever
        fleet "a fleet" 300
        fleet "a fleet" 1000

    event
        system
            remove
            fleet 1000
```

It'll remove `fleet "a fleet" 300` since that is the first one with the requested fleet name.

My method reduces the confusion.

Reply:
Why not use the UUID we use for missions and ships?

Reply:
The ability to specify custom ids critical to the feature.

Reply:
If you think "id" is a confusing name, I can rename it, but "name" is already taken. I could use "tag" but usually "tag" is a set of things rather than a single one.

Regardless of what it is called, it is critical to the feature.